src/debuggability.js

```diff
@@ -154,12 +154,17 @@ var fireDomEvent = (function() {
             var event = new CustomEvent("CustomEvent");
             util.global.dispatchEvent(event);
             return function(name, event) {
-                var domEvent = new CustomEvent(name.toLowerCase(), {
+                var promise = event.promise;
+                var reason = event.reason;
+                var eventData = {
                     detail: event,
-                    cancelable: true,
-                    reason: event.reason,
-                    promise: event.promise
+                    cancelable: true
+                };
+                Object.defineProperties(eventData, {
```

Comment:
If you're using this syntax anyway - can you please use the helpers from util that work in ES3 environments for consistency?

Reply:
Done (I think!)